### ISSUE TITLE
Request: rearrange shopping tab to be more user friendly / easier to use quickly

### ISSUE THREAD

**Issue by Rezy942** [Auctionator/Auctionator#1318]
**Is your feature request related to a problem? Please describe.**
Current shopping list tab is good, but could be designed a bit better to more user friendly by eliminating unnecessary clicks each time you open auction house as the shopping lists are hidden inside a drop down menu, which feels unnecessary. It gets very annoying when opening auction house constantly as AH goblin. Creating a shopping list as a player new to this addon is also very confusing, which wasnt the case few expansions ago before AH overhaul.

**Describe the solution you'd like**
See screenshot below
1) remove shopping list dropdown menu
2) move this function to the left as a tab, and make it default tab when you open shopping lists. This tab will show a list of your shopping lists, preferably with sorting configurations so you can move them up and down as needed. This ensures player simply opens auction and instantly clicks on shopping list they want, eliminating an extra click.
3) Keep other 2 tabs, but push them to the right. Rename Shopping List to "items" or something as thats what it is - items inside a shopping list.
![image](https://user-images.githubusercontent.com/122007087/210778165-88c7280e-eefb-4f3e-924b-8fe3357406b7.png)

Thank you so much for amazing addon and all the work you do. I hope you will consider this as it definetely annoys me, AH goblin, with many different shopping lists. I cannot choose 1 to be the default.

**Comment by Rezy942** [comment]
Thank you so much for replying so quick and so in depth. I like your ideas (mine were just quick thoughts based on some minor annoyances). If you do decide to change things up, I will leave it in your capable hands to design it for the better.

Have a good one :)

**Comment by plusmouse** [comment]
For now going to leave the Shopping tab as is.

**Comment by Rezy942** [comment]
Again, too bad. Hope you reconsider in the future.
Most things shopping list related are designed awfully.

**Comment by plusmouse** [comment]
Going to reopen this as I've ended up working on a shopping list internals rework anyway that would make this easier.

**Comment by plusmouse** [comment]
Work in progress new shopping tab design (only got the list selection so far, and its missing the edit/search buttons):
![list sample ui 3](https://github.com/Auctionator/Auctionator/assets/60280559/a52fe7d7-bbef-4ff8-908a-6c115e06adc7)

# Proposed design

![image](https://github.com/Auctionator/Auctionator/assets/60280559/3537e297-f8a6-44e0-9048-044842a75c55)

I haven't figured out where to put recent searches yet though.

**Comment by Rezy942** [comment]
You are amazing! Wow! Looks much better already, cant wait :)
Perhaps recent searches under export/import? As thats technically the least used feature (for me).

Also on a separate note as I opened this suggestion before, with this shopping list rework, is there any possibility or room to add some sort of notes regarding item quantity needed to purchase?

Use case example:
I got the soulbound item to make Chip pet in Dragonflight expansion.
What do you do then? Go wowhead and search comments for how many rock cluster and element blood you need to buy the pet, to later sell it on AH: https://www.wowhead.com/item=199326/chip#comments
Then, you write it down or memorize, and go to buy on AH. Now i can make shpping list for it (as its not rare drop, i dont want to search wowhead each time...) but the list wont remind me quantities.
So in the end yes shopping list works for me, BUT i still rely on memory for quantities, and it sucks!

Suggested way to fix this: allow a "note" field for every item entry, which will appear when you hover over the materials under each list. In your screenshot it's Monelite Ore for example.

Just my 2 cents thank you :)

**Comment by plusmouse** [comment]
Visual update:
![image](https://github.com/Auctionator/Auctionator/assets/60280559/6b7882fd-edee-4ca6-b6f3-672b1ae833b9)

About the other suggestion, I haven't forgotten it, and there's been a few other requests for quantity saving, including auto-selection of the quantity when purchasing. What I said before in #1320 still stands, and if it its added it will be in conjunction with features to prepopulate the quantity field.

**Comment by plusmouse** [comment]
Just to clarify, closing a feature request doesn't always mean "never" sometimes it is just "not at this time" - which is affected by how busy I am.

**Comment by Rezy942** [comment]
No worries - in the end the addon reflects your vision. Whether it gets added or not, i just wanted to provide my feedback as sometimes author's use case varies dramatically compared to other users :)

**Comment by plusmouse** [comment]
Got a preview of the list selection and search working:
[NewShoppingTab-1b02a07.zip](https://github.com/Auctionator/Auctionator/files/11523979/NewShoppingTab-1b02a07.zip)
![image](https://github.com/Auctionator/Auctionator/assets/60280559/c56cc37e-da4f-4064-8580-99fbea4055bf)
If you want you can try the Shopping V2 tab in the AH to see how its going so far.

This is currently a separate addon to install (just for development, so that I don't need to break Auctionator to test the new design)

**Comment by plusmouse** [comment]
Yeah, this addon replaces the main Auctionator addon. The build attached for for retail wow, if you're trying it on classic then it won't work. These are the versions for the different editions of WoW:
[Retail](https://github.com/Auctionator/Auctionator/files/11540616/Auctionator-10.1.2-13-g47b8c82.zip)
[Wrath](https://github.com/Auctionator/Auctionator/files/11540618/Auctionator-10.1.2-13-g47b8c82-wrath.zip)
[Vanilla](https://github.com/Auctionator/Auctionator/files/11540619/Auctionator-10.1.2-13-g47b8c82-classic.zip)

**Comment by Rezy942** [comment]
Working good so far, wanted to share 1 feedback to greatly improve QoL and speed up adding new searches to shopping lists:

Allow Shift + left click to copy item name and add it to "search term" bar. This behavior is similar to blizzard.
A lot of the time you dont know exact name or too lazy to write out the entire thing. As u see in screenshot below, i wrote half of it, but want to fill in the rest by shift clicking an item. 
This is already useful for 1 item with long name, but imagine I want to add about 10 items from this list.
I can quickly shift click each item and hit "add to list", quickly populating it without rewriting every single item by hand.

Current behavior: normal or shift clicking an item always takes it to default buy tab page.

![image](https://github.com/Auctionator/Auctionator/assets/122007087/fc322623-0444-4978-bb40-786ca8552e07)

**Comment by Rezy942** [comment]
Minor feedback #2:

Currently when you click on list, it opens up and instantly searches entire list, in this case list "pets" on screenshot below.
Now, lets say user has many many battle pets (or any other items) in that list, and wants to search one specific item instead.
What happens is addon triggers one big search, then user can select what he actually wants, and another search goes out.
This might be an issue for some users when it comes to loading times of the "big" list, as you cannot search anything else until that loading is complete. The list i have below has a short delay before you can search what u want, due to loading time lag.
I imagine it will get worse if u add another few terms.

Possibly simple solution:
Make the - / + sign to the left of item name a separate clickable entity, that does NOT trigger a search.
The item name can stay as-is, if u click, it triggers a search and opens/closes the list.
This will allow users to decide if they want to simply close or open a list without triggering another search :)


Example:
![image](https://github.com/Auctionator/Auctionator/assets/122007087/e2e7b63c-fe99-432a-bf13-89a6ff0fb65d)

**Comment by plusmouse** [comment]
1. Try this version:
[Auctionator-10.1.2-17-gd34c939.zip](https://github.com/Auctionator/Auctionator/files/11542610/Auctionator-10.1.2-17-gd34c939.zip)

3. You can turn the automatic list search off in the settings
![image](https://github.com/Auctionator/Auctionator/assets/60280559/bba3a6cf-d800-44e0-8c5c-6126d02f4f94)

**Comment by Rezy942** [comment]
Tested just now - works amazing! I really appreciate you adding this feature in. Adding items is SO EASY now!
I will keep messing around with it in case theres a bug, but if its complete, i'd assume it should be ready for release :)

**Comment by plusmouse** [comment]
That's great 🎉 

I've put it into the alpha releases now for final testing before a full release.
